src/Blocks/BlockButton.h

```diff
@@ -113,15 +122,36 @@ class cBlockButtonHandler :
 		}
 	}
 
+	static bool CanBePlacedOn(BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, eBlockFace a_BlockFace)
```

Comment:
Since button place-ability behavior is specific to the target block, we need a function akin to what's seen in [BlockTorch.h](https://github.com/csunday95/cuberite/blob/2ccaa5eb9f41e166d18d24736b9242839f48bc12/src/Blocks/BlockTorch.h#L88)